rom/dev/src/flow/cold_reset/fmc_alias.rs

```diff
@@ -313,6 +313,27 @@ impl FmcAliasLayer {
             manifest.fmc.load_addr,
             manifest.fmc.size
         );
+        cprintln!(
+            "[afmc] Loading Runtime at address 0x{:08x} len {}",
+            manifest.runtime.load_addr,
+            manifest.runtime.size
+        );
+
+        // Detect if FMC and RT address ranges overlap
+        // Load address of FMC <= End address of RT
+        // and End address of FMC >= Load address of RT
+        if manifest.fmc.load_addr <= manifest.runtime.load_addr + manifest.runtime.size
```

Comment:
Can we add an integration test?

Reply:
Coordinate with @ajisaxena so you don't duplicate work

Reply:
Actually, let's add the integration test in a future PR. Updated #350 with the new task.